src/resourcetiming-compression.js

```diff
@@ -803,6 +804,23 @@
         return this.optimizeTrie(this.convertToTrie(results), true);
     };
 
+    // eslint-disable-next-line max-len
+    ResourceTimingCompression.HOSTNAME_REGEX = /^(?:(?:(([^:\/#\?]+:)?(?:(?:\/\/)(?:(?:(?:([^:@\/#\?]+)(?:\:([^:@\/#\?]*))?)@)?(([^:\/#\?\]\[]+|\[[^\/\]@#?]+\])(?:\:([0-9]+))?))?)?)?((?:\/?(?:[^\/\?#]+\/+)*)(?:[^\?#]*)))?(\?[^#]+)?)(#.*)?/;
```

Comment:
All constants should be up at the top.

This regex is insane!  It looks like this is breaking up the entire URL, so you get capture groups for everything:

![image](https://cloud.githubusercontent.com/assets/1004649/23468638/48b37df8-fe6f-11e6-9d1a-be3302864e1f.png)

Can we get a simpler regex that just matches the domain?

```
https?:\/\/([^\/]+)
```

![image](https://cloud.githubusercontent.com/assets/1004649/23468808/d0093310-fe6f-11e6-94ef-f5bd99454448.png)